src/main/recorder.ts

```diff
@@ -192,58 +241,53 @@ type RecorderOptionsType = {
             const isRaid = metadata.category == VideoCategory.Raids;
             const isLongEnough = (metadata.duration - overrun) >= this._options.minEncounterDuration;
 
-            if ((!isRaid || isLongEnough) && !discardVideo) {
-                // Cut the video to length and write its metadata JSON file.
-                // Await for this to finish before we return to waiting state.
-                await this.finalizeVideo(metadata, outputFilename);
-            } else {
+            if (isRaid && !isLongEnough) {
                 console.info("[Recorder] Raid encounter was too short, discarding");
+            } else if (discardVideo) {
+                console.info("[Recorder] Discarding video");
+            } else {
+                const bufferFile = obsRecorder.getObsLastRecording();
+                if (bufferFile) {
+                    await this.queueVideo(bufferFile, metadata);
+                } else {
+                    console.error("[Recorder] Unable to get the last recording from OBS. Can't process video.");
+                }
             }
 
-            // Run the size monitor to ensure we stay within size limit.
-            // Need some maths to convert GB to bytes
-            runSizeMonitor(this._options.storageDir, this._options.maxStorage)
-                .then(() => {
-                    if (mainWindow) mainWindow.webContents.send('refreshState');
-                });
-
-            // Clean-up the temporary recording directory. 
-            this.cleanupBuffer(1);
-
             // Refresh the GUI
             if (mainWindow) mainWindow.webContents.send('refreshState');
 
             // Restart the buffer recording ready for next game.
-            setTimeout(async () => {
-                this.startBuffer();
-            }, 5000)
+            this.startBuffer();
```

Comment:
This makes me nervous. I think I've spent several afternoons scratching my head around this and found that if we don't wait at least some time between stopping and starting OBS things go wrong. 

I think previously it was 2 seconds, for some reason I changed it to 5 seconds. But this makes it 0 seconds. I think the symptom was the signalling going wrong. Not suggesting you change this immediately, but we'll need to carefully test this.

Worst case we can revert this change and the PR still massively improves things in this area anyway.

Reply:
We are probably indirectly waiting at least some time given the earlier await on the adding to queueing that I commented on above. If we need this to function though, this might be a bit fragile.

Reply:
I will revert this - I removed it because all the logic that was depending on it has been removed/deprecated.

However, restarting the buffer here isn't any different from when it restarts every 5 minutes of idling, is it?